opsdroid/connector/matrix/connector.py

```diff
@@ -414,45 +448,50 @@ async def _send_room_creation(self, creation_event):
     @register_event(events.RoomName)
     @ensure_room_id_and_send
     async def _send_room_name_set(self, name_event):
-        return await self.connection.set_room_name(name_event.target, name_event.name)
+        return await self.connection.room_put_state(
+            name_event.target, "m.room.name", name_event.name
+        )
 
     @register_event(events.RoomAddress)
     @ensure_room_id_and_send
     async def _send_room_address(self, address_event):
-        try:
-            return await self.connection.set_room_alias(
-                address_event.target, address_event.address
-            )
-        except MatrixRequestError as err:
-            if err.code == 409:
+        res = await self.connection.room_put_state(
+            address_event.target, "m.room.aliases", address_event.address
+        )
+
+        if isinstance(res, nio.RoomPutStateError):
+            print("Error setting room address: " + res.message)
```

Comment:
The previous behaviour here seems wrong. I think we should raise the exception unless it's a `409` in which case we should log and then silently continue.